examples/webgl_interactive_cubes_gpu.html

```diff
@@ -68,17 +69,66 @@
 				scene = new THREE.Scene();
 				scene.background = new THREE.Color( 0xffffff );
 
-				pickingScene = new THREE.Scene();
-				pickingTexture = new THREE.WebGLRenderTarget( 1, 1 );
-
 				scene.add( new THREE.AmbientLight( 0x555555 ) );
 
 				const light = new THREE.SpotLight( 0xffffff, 1.5 );
 				light.position.set( 0, 500, 2000 );
 				scene.add( light );
 
-				const pickingMaterial = new THREE.MeshBasicMaterial( { vertexColors: true } );
-				const defaultMaterial = new THREE.MeshPhongMaterial( { color: 0xffffff, flatShading: true, vertexColors: true, shininess: 0	} );
+				const defaultMaterial = new THREE.MeshPhongMaterial( {
+					color: 0xffffff,
+					flatShading: true,
+					vertexColors: true,
+					shininess: 0
+				} );
+
+				// set up the picking texture to use a 32 bit integer so we can write and read integer ids from it
+				pickingScene = new THREE.Scene();
+				pickingTexture = new THREE.WebGLRenderTarget( 1, 1, {
+
+					type: THREE.IntType,
+					format: THREE.RedIntegerFormat,
+					internalFormat: 'R32I',
+
+				} );
+				const pickingMaterial = new THREE.ShaderMaterial( {
+
+					glslVersion: THREE.GLSL3,
+
+					vertexShader: /* glsl */`
+						attribute int id;
+						flat varying int vid;
+						void main() {
+
+							vid = int( id );
+							gl_Position = projectionMatrix * modelViewMatrix * vec4( position, 1.0 );
+
+						}
+					`,
+
+					fragmentShader: /* glsl */`
+						layout(location = 0) out int out_id;
+						flat varying int vid;
+
+						void main() {
+
+							out_id = vid;
+
+						}
+					`,
+
+				} );
+
+				function applyId( geometry, id ) {
+
+					const position = geometry.attributes.position;
+					const array = new Int32Array( position.count );
+					array.fill( id );
```

Comment:
Once #21606 is merged I would like to change this to an Int16 array and use the new "integer" declaration field here.